src/tribler-core/tribler_core/modules/watch_folder.py

```diff
@@ -41,14 +41,11 @@ def check_watch_folder(self):
             return
 
         # Make sure that we pass a str to os.walk
-        watch_dir = self.session.config.get_watch_folder_path().to_text()
+        watch_dir = str(self.session.config.get_watch_folder_path())
 
         for root, _, files in os.walk(watch_dir):
-            root = path_util.Path(root)
-            for name in files:
-                if not name.endswith(".torrent"):
-                    continue
-
+            root = Path(root)
+            for name in (name for name in files if name.endswith(".torrent")):
```

Comment:
That's a *bad* idea. As you can see, `files` is produced by the `os.walk` **iterator**. You don't really want to build intermediate tuples with that, as these are gonna be HUGE.